include/dfu/dfu_target.rst

```diff
@@ -0,0 +1,48 @@
+.. _lib_dfu_target:
+
+DFU target
+##########
+
+The DFU target library provides a common API for different types of firmware upgrades, for example, an MCUboot style upgrade or a modem firmware upgrade.
+Use this library in your component to do different types of firmware upgrades against a single interface.
+
+When initializing the DFU target library, you must provide information about the type of firmware upgrade.
+To do so automatically, send the first fragment of the firmware to the function :cpp:func:`dfu_target_img_type`.
+This function can identify all supported firmware upgrade types.
+The result of this call can then be given as input to the :cpp:func:`dfu_target_init` function.
+
+
+Supported DFU targets
+*********************
+
+Every supported DFU target must implement the set of functions defined in :file:`subsys/dfu/src/dfu_target.c`.
+
+The following sections describe the DFU targets that are currently supported.
+
+MCUboot style upgrades
+======================
+
+This type of firmware upgrade writes the data given to the :cpp:func:`dfu_target_write` function into the secondary slot specified by MCUboot's flash partitions.
+
+When the complete transfer is done, call the :cpp:func:`dfu_target_done` function to mark the firmware that is stored in the secondary slot as ready to be booted.
+On the next reboot, the device will run the new firmware.
+
+Modem firmware upgrades
+=======================
+
+This type of firmware upgrade opens a socket into the modem and passes the data given to the :cpp:func:`dfu_target_write` function through the socket.
+The modem stores the data in the memory location for firmware patches.
```

Comment:
Is it correct that the library passes the data, the modem stores it?
And is it the modem or the library that deletes the old patch?

Reply:
Yeah data is only passed to the modem firmware, the modem firmware is responsible of storing it.

The library requests the modem to delete the old patch.